### PR TITLE
Is Computational Storage a Newfound Solution to Edge Storage Problems?

### LINKED ISSUE #4309: [Edge Computing] Is Computational Storage a Newfound Solution to Edge Storage Problems?
#### Proposed title of article
Is Computational Storage a Newfound Solution to Edge Storage Problems?

#### Proposed article introduction
With the growth of data generation at the Edge and the need to get value from that data quickly, the market has run into a hurdle on how to get enough compute and processing with the available space, power, and budget. The ability to deploy compute resources within the storage devices with Computational Storage is key to the growth of this market.

Computational storage is all about making storage devices smarter to process the data directly where it is stored. This approach reduces the movement of large amounts of data to external processing and delivers myriad benefits, such as reduced latency and bandwidth usage, increased security, and energy savings.

Applying computational storage is critical to address the real-time processing requirements of many machine learning (ML) or analytics applications, and other use cases from IoT to edge computing.

#### Key takeaways 

- Why computational storage makes sense for the edge.
- How computational storage drives can best serve the needs of data centers.
- How computational storage solves edge storage issues.

#### Article quality
Most of the available articles are light in content - they define computational storage and discuss its use cases. But, in this article, we will discuss how computational storage is interlinked with edge storage, how it is serving data centers and the storage issues it is solving for edge devices.
#### References
N/A

### PR DESCRIPTION
**Pull request by BridgetMwikali**
this closes #4309

### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [ ] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [ ] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #issue-number`

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by Kayere** [comment]
Plagiarism check done ✔️

File: content/articles/is-computational-storage-a-newfound-solution-to-edge-storage-problems/index.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+A firm that implements computational storage at the edge experiences a better return on investment. Its teams become more efficient because of quick data processing at the firm. It obtains real-time data analytics quickly. Such information enables it to undertake an effective decision-making process and eliminates avoidable errors.
+
+### Further reading 
+- [The Evolution of Computing to Edge Computing](/engineering-education/what-is-edge-computing/)
+
+- [Introduction To Cloud Computing](/engineering-education/introduction-to-cloud-computing/)
+
+- [Edge Computing Use Cases Driving Innovation](/blog/edge-compute-use-cases/)
+
+- [7 Predictions in Edge Computing for 2021](/blog/7-edge-computing-predictions-2021/)
```
Comment (Bayler) [inline content/articles/is-computational-storage-a-newfound-solution-to-edge-storage-problems/index.md:70]: Please ensure that all your links are working. The current URLs display a `file not found` error

File: content/articles/is-computational-storage-a-newfound-solution-to-edge-storage-problems/index.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+For instance, edge computing makes it easy to process crime reports using surveillance camera images. Such data can undergo processing to reveal geometric figures instantly. This kind of flexibility ensures that data processing occurs at any point and time.
+
+The flexibility that comes with computation at the edge enhances performance. It ensures that data arrives on time and in a stable form. So, there is no risk of losing crucial computational information, making edge computing a reliable option.
+
+Computational storage is a step closer to cyber-security and greater privacy. Computation at the edge enables firms to limit their usage of [cloud computing](/engineering-education/introduction-to-cloud-computing/). With computational storage at the edge, you can process much of the data without connecting to the internet. Edge computing devices are designed such that they can perform several functions offline. In this way, firms only incorporate internet services like cloud computing if necessary. Ultimately, they reduce the risk of becoming victims of cybersecurity attacks.
```
Comment (Bayler) [inline content/articles/is-computational-storage-a-newfound-solution-to-edge-storage-problems/index.md:29]: Please ensure that you add the correct link to the cloud computing resource. The link above seems incomplete

File: content/articles/is-computational-storage-a-newfound-solution-to-edge-storage-problems/index.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+Adopting edge computing allows quicker processing of data as it comes in. This is why platforms like [Apache Hadoop](https://hadoop.apache.org/) and others that have adopted computational storage are gaining popularity. Computational storage efficiently deals with the large and quick influx of data. So, there is no need to have more extensive data centers using traditional computation models.
+
+Besides, computation storage enhances flexibility in the data processing. If you have to collect data at one point and later process it somewhere else, you need to operate close to the processors. This scenario imposes limitations on your ability to create value from data when distant from processors. Computation at the edge allows you to collect and process data instantly.
+
```
Comment (Bayler) [inline content/articles/is-computational-storage-a-newfound-solution-to-edge-storage-problems/index.md:24]:
```suggestion
Besides, computation storage enhances flexibility in data processing. If you have to collect data at one point and later process it somewhere else, you need to operate close to the processors. This scenario imposes limitations on your ability to create value from data when distant from processors. Computation at the edge allows you to collect and process data instantly.
```

File: content/articles/is-computational-storage-a-newfound-solution-to-edge-storage-problems/index.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+#### Ensuring efficiency in hyper-scale data centers
+
+Adopting edge computing paves the way for efficiency in hyper-scale data centers. Firms, such as [Facebook](https://about.facebook.com/), have large data centers that deal with a massive data influx. The ability of these data centers to run efficiently depends on their ability to optimize physical space and resource utilization. They must also [minimize power and cooling needs](https://storageswiss.com/2019/04/04/use-cases-for-computational-storage/). Constantly relying on the traditional model of data processing would derail data processing. Considering the large amount of data coming in and leaving, it would create a time lag. Data centers must, therefore, adopt edge computing to remove these unnecessary processing hitches. Edge computing transforms them into efficient data processing centers.
```
Comment (Bayler) [inline content/articles/is-computational-storage-a-newfound-solution-to-edge-storage-problems/index.md:39]:
```suggestion
Adopting edge computing paves the way for efficiency in hyper-scale data centers. Firms like [Facebook](https://about.facebook.com/) have large data centers that deal with a massive data influx. 
```

File: content/articles/is-computational-storage-a-newfound-solution-to-edge-storage-problems/index.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
+Is Computational Storage a Newfound Solution to Edge Storage Problems?
+
+In an era of increased data flow, everyone wants to get value from processed data quickly. Yet edge storage delays the processing because of an existing gap between data storage and processing. It is increasingly important to bridge this gap to improve data processing speeds.
+
+Computational storage is key to bridging that gap. It enables storage devices to process data directly at the point of storage. It reduces data overload at the edge and improves processing speeds. Applying computer storage reduces real-time data processing challenges.
+
+This article will discuss how computational storage is interlinked with edge storage, how it is serving data centers and the storage issues it is solving for edge devices.
```
Comment (Bayler) [inline content/articles/is-computational-storage-a-newfound-solution-to-edge-storage-problems/index.md:7]:
```suggestion
This article will discuss how computational storage is interlinked with edge storage, how it is serving data centers, and the storage issues it is solving for edge devices.
```

**Comment by Bayler** [comment]
@BridgetMwikali Hi. This was a nice read. Keep up the good work. Just some minor revisions before we can move on for final review.

**Comment by Bayler** [comment]
@BridgetMwikali Please work on the revisions.

File: content/articles/is-computational-storage-a-newfound-solution-to-edge-storage-problems/index.md (hunk 1 above)
Comment (BridgetMwikali) [reply content/articles/is-computational-storage-a-newfound-solution-to-edge-storage-problems/index.md:70]: Hi @Bayler These links are from Section.io. There is no need of including https://www.section.io/ . They'll work once the article is published.

Comment (Bayler) [reply content/articles/is-computational-storage-a-newfound-solution-to-edge-storage-problems/index.md:69]: @BridgetMwikali I've just confirmed. It's alright.

**Comment by hectorkambow** [comment]
jumping on this one

**Comment by hectorkambow** [comment]
Overall very nice work 🚀

**Pull request merged**